Argus-Labs/world-engine
# feat(cardinal): Create cardinal package that only exposes the portions of the world e…

…ngine that game developers need to use

Closes: CAR-110, CAR-140

## What is the purpose of the change

- Reduce the methods that are exposed by the ecs, and sub packages.
- Reduce the number of packages that are imported when creating a Cardinal based game.
- Put all godoc information into a single package to make it easier to discover/process the Cardinal package.
- Create example_transactiontype_test.go, which demonstrates how to write sample code that can be associated with specific Cardinal methods.

## Testing and Verifying

The best way to verify this PR is to look at the generated godoc for this cardinal package. 

- `git checkout` this branch
- run `godoc` from the `world-engine/cardinal` directory
- navigate to http://localhost:6060/pkg/pkg.world.dev/world-engine/cardinal/

Take note of the Example section of `type TransactionType`
